packages/entities/exercise-groups/missing-letter.ts

```diff
@@ -38,13 +38,21 @@ export class MissingLetter extends AbstractExerciseGroup {
         const wordLetters = (wordObj ? wordObj.toString() : word).split('');
         const numberOfOptions: number = 3;
         const numberMissing: number =
-          this.props.difficulty < 0.2
+          word.length === 2
             ? 1
-            : this.props.difficulty < 0.4
-              ? 2
-              : this.props.difficulty < 0.6
-                ? 3
-                : this.props.difficulty < 0.8 ? 4 : wordLetters.length;
+            : this.props.difficulty < 0.2
+              ? 1
+              : this.props.difficulty < 0.4
+                ? 2
+                : word.length === 3 && this.props.difficulty < 0.6
+                  ? 2
+                  : this.props.difficulty < 0.6
+                    ? 3
+                    : word.length === 4 && this.props.difficulty < 0.8
+                      ? 3
+                      : this.props.difficulty < 0.8
+                        ? 4
+                        : wordLetters.length - 1;
```

Comment:
This code is hard to understand, maybe we can refactor it. E.g. something like 
```javascript
const numberMissing = 
  this.props.difficulty >= 0.8
    ? wordLetters.length -1
    : Math.min(wordLetters.length - 1, Math.ceil(this.props.difficulty*5)) 
```
or maybe extract the old code into a function `getMissingByDifficulty(difficulty)` and then write:
```javascript
const numberMissing = Math.min(wordLetters.length - 1, getMissingByDifficulty(this.props.difficulty)) 
```